plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -464,15 +464,16 @@ public Optional<JoinApplicationResult<ConnectorTableHandle>> applyJoin(
         int nextSyntheticColumnId = max(leftHandle.getNextSyntheticColumnId(), rightHandle.getNextSyntheticColumnId());
```

Comment:
I can see more than one change here:
- promote the oracle change to other JDBC connectors
- allow changing the limit per connector

It's hard for me to understand from the context which one is a bug and which one is a feature. Do you think it would make sense to split commits so that it's easier to understand where one business change is done and the other starts?

Reply:
Fair point. I'll look at this again.

Reply:
split into two commits